Added Meld Implementation, proper asset order and other optimizations

Zorkats opened the pull request:
Merge Tool Integration

Added Meld merge tool support with automatic detection across Windows and Linux
Implemented 3-way merge capability for resolving conflicts between multiple mods
Added fallback mechanisms when merge tools are unavailable
Enhanced user warnings about missing merge tools and their impact

New Object Handling

Added `HandleNewObjects()` function to detect objects added by mods that don't exist in vanilla
Implemented` importWithNewObjects() `for proper handling of new content during import
Enhanced object detection logic with proper filename parsing for GameMaker object events

Performance Improvements

Replaced nested loop file comparison with Dictionary-based lookup for O(1) file matching
Optimized file processing in `CompareCombine()` function
Reduced redundant file system operations

Enhanced Asset Order Management

Improved AssetOrder.txt merging using Meld for proper sprite ordering
Added conflict detection for asset ordering changes
Implemented fallback logic for asset order merging when merge tools unavailable

Further improvements to overall system stability, system security and other minor adjustments have been made.

techy804 reviewed (changes requested):
<img width="1190" height="409" alt="{1838F3FC-F0A7-44F2-9B9C-DCF4A3247158}" src="https://github.com/user-attachments/assets/6e62c5fa-813f-4fcc-a9a5-26831c321223" />

Zorkats commented:
1. Replaced Meld with Git for Code Merging

Before: Simple 3-way merge with Meld that could only handle 2 mods
After: Git octopus merge supporting unlimited mods with automatic conflict resolution
Preserves changes from all mods instead of last-mod-wins

2. Complete File Tracking System

Before: Only tracked files that existed in vanilla
After: Comprehensive tracking of ALL files across vanilla and all mods
New ModFileInfo class for better organization
Files are properly categorized and handled based on their type

3. Intelligent AssetOrder.txt Merging

Before: Simple append/overwrite causing sprite animation issues
After: Context-aware merging that:

Preserves vanilla asset order
Places new assets near related ones
Validates animation sequences remain together
Detects and reports ordering conflicts

4. PNG Sprite Validation System

Added PNG header validation (8-byte signature check)
Smart sprite selection when multiple mods modify the same sprite
Falls back to vanilla when mod sprites are corrupt
Post-merge validation with detailed reporting

Technical Improvements
Code Quality Improvements:

Modular design with single-responsibility methods
Comprehensive error handling and null checks
Detailed logging at every decision point
File content verification after operations
Fallback strategies for all failure cases


Sprite animations remain intact
All mod changes are preserved where possible
Corrupt files are detected and reported
Detailed logging shows exactly what happened and why
Validation reports help identify issues

 Performance Considerations

File size may increase due to texture page duplication (documented issue with workaround suggestions)
Slightly longer processing time due to validation checks (worth it for reliability)
Git operations add ~2-3 seconds per merge (negligible compared to overall process)

 Testing
Tested with:

Single mod modifications
Multiple mods modifying same files
Mods adding new objects
Sprite-heavy mods with complex animations
Edge cases (empty files, missing vanilla, corrupt PNGs)
